### DISCUSSION TITLE
What is the versioning policy on sentry-dotnet?

### DISCUSSION THREAD

**Discussion by jbgh2** [getsentry/sentry-dotnet#1136] [Q&A] [answered]
Basically, is it safe to assume that 3.* will be backwards compatible?

**Comment by bruno-garcia** [comment, marked as answer]
That's the goal even though we've shipped a few small breaking changes on parts of the API which are not really expected to affect users. But technically breaking.

We ship majors when we have clear breaking changing in API or behavior (like sending session data by default).
Minor when new features are added.
Patch for fixes.

This is more of a goal than a hard rule. We just shipped a patch that has a new package `Sentry.Tunnel` but since this isn't something that adds value to users of any package ([it's a middleware to tunnel events through](https://docs.sentry.io/platforms/javascript/troubleshooting/#dealing-with-ad-blockers)), it didn't get a minor bump.

**Reply by jbgh2** [reply]
Thank you for the update, I had assumed it was something like this but couldn't find any documentation.

**Reply by bruno-garcia** [reply]
We maintain migration pages on SDK docs, such as this one: https://docs.sentry.io/platforms/dotnet/migration/

Any steps you'd need to do to move from one major to another should be laid out there.

**Comment by mattjohnsonpint** [comment]
Just to add to this, we generally try to follow https://semver.org/
